spec/mem/tree_borrows/state_machine.md

```diff
@@ -44,49 +49,70 @@ impl Permission {
     fn child_read(self) -> Result<Permission> {
         ret(
             match self {
-                Permission::Reserved => Permission::Reserved,
-                Permission::Active => Permission::Active,
-                Permission::Frozen => Permission::Frozen,
                 Permission::Disabled => throw_ub!("Tree Borrows: reading from the child of a pointer with Disabled permission"),
+                _ => self,
             }
         )
     }
 
-    fn child_write(self) -> Result<Permission> {
+    fn child_write(self, protected: Protected) -> Result<Permission> {
         match self {
-            Permission::Reserved => ret(Permission::Active),
-            Permission::Active => ret(Permission::Active),
+            Permission::Reserved { conflicted: true } if protected.yes() =>
```

Comment:
Why do some transitions check `protected.yes()` and others `protected.active(accessed)`?

Reply:
A foreign read should make a non-accessed Reserved becomes `Reserved { conflicted: true }`.

Reply:
Explicitly splitting these two states is somewhat clear to me, what do you think?

Reply:
> Why do some transitions check `protected.yes()` and others `protected.active(accessed)`?

For reasons outlined [here](https://github.com/minirust/minirust/pull/221#issuecomment-2276884520).

Reply:
Miri achieves this by doing this here: https://github.com/rust-lang/miri/blob/master/src/borrow_tracker/tree_borrows/tree.rs#L111

Reply:
Yeah I think I'd prefer if we did what Miri does, and have a single "transitions to disabled on initialized protected locations are UB" check.